### DISCUSSION TITLE
Aborting failing operation early (CreateTeamAsync)

### DISCUSSION THREAD

**Discussion by web265p3** [pnp/pnpcore#1319] [General]
I can add a new Team to an existing group, with the following line:

        var ctx = await _pnPContextFactory.CreateAsync(new Uri("https://contoso-admin.sharepoint.com"));
        var contextWithTeam = await ctx.GetTeamManager().CreateTeamAsync(new TeamForGroupOptions(new Guid("xxxxxxx-91ae-4d30-9324-2d2854bb9407")));

This works fine, but what if the Team already exists?
In this case my function stucks in a loop that breaks after 10 Minutes:
```

warn: PnP.Core.Services.PnPContext[0]
      Following exception was eaten while creating a Team for group xxxxb3de2-91ae-4d30-9324-2d2854bb9407 : HttpResponseCode: 409
      Code: Conflict
      Message: Team already exists
      ClientRequestId: 3fa8d11a-da92-4971-82d0-40afb48caafb
      message: Team already exists
      code: Conflict
      innerError: {}
      date: 2023-11-20T09:35:35
      client-request-id: 3fa8d11a-da92-4971-82d0-40afb48caafb
```

If a "Team already exists" its obvious that trying further does not make any sense. It would be better if the loop directly was aborted. How can I archive this? TeamForGroupOptions does not provide any more options.

**Comment by jansenbe** [comment]
@web265p3 : I've added fix for this, see https://github.com/pnp/pnpcore/commit/9c5e4f38128e3b78c484ecf1e3bba0410aada4dd. Please give this a try with the next nightly build

**Comment by web265p3** [comment]
@jansenbe Perfect, you are the best! It works ;-)
